### PR TITLE
Update aws rds how to guide

### PR DESCRIPTION
**Pull request by teddyking**
## About

This PR updates the how-to guide for configuring dynamic provisioning of AWS RDS PostgreSQL instances with the following changes:

* Ensure the APIs used in the `Composition` match those provided by the `Provider` that we recommend
* Recommend and make use of AWS Provider Family (previously the more monolithic AWS Provider)
* Also updated the know limitation related to degradation of performance with a link to Provider Families 

Which other branches do you want a technical writer to cherry-pick this PR to (if any)? N/A

### CODE REVIEW

File: services-toolkit/how-to-guides/dynamic-provisioning-rds.hbs.md [hunk 1]
```diff
@@ -33,23 +33,67 @@ To configure dynamic provisioning for AWS RDS service instances, you must:
...
+    metadata:
+      name: upbound-provider-aws-rds
+    spec:
+      package: xpkg.upbound.io/upbound/provider-aws-rds:v0.36.0
+      controllerConfigRef:
+        name: upbound-provider-family-aws
+    ---
+    # The ControllerConfig applies settings to a Provider Pod.
+    # With family Providers each Provider is a unique Pod running in the cluster.
+    apiVersion: pkg.crossplane.io/v1alpha1
```
Comment (Samze) [inline services-toolkit/how-to-guides/dynamic-provisioning-rds.hbs.md:78]: **thought**

I get 
```
Warning: This API is deprecated and is scheduled to be removed in a future release.
controllerconfig.pkg.crossplane.io/upbound-provider-family-aws created
``` 

But I recall that they actually had no plans to deprecate this as there is no alternative. But is this still their recommended model to use provider families?

File: services-toolkit/how-to-guides/dynamic-provisioning-rds.hbs.md [hunk 2]
```diff
@@ -33,23 +33,67 @@ To configure dynamic provisioning for AWS RDS service instances, you must:
...
+
+1. Apply the file to the Tanzu Application Platform cluster by running:
+
+    ```console
+    kubectl apply -f provider-family-aws.yaml
+    ```
+
+1. Confirm that both Providers have been installed successfully by running `kubectl get providers` and checking for INSTALLED=True and HEALTHY=TRUE.
+
+Next you must create a `ProviderConfig` for the Providers. Refer to the sections titled "Create a Kubernetes secret for AWS" and "Create a ProviderConfig" in the [Upbound documentation](https://marketplace.upbound.io/providers/upbound/provider-aws/v0.36.0/docs/quickstart).
```
Comment (Samze) [inline services-toolkit/how-to-guides/dynamic-provisioning-rds.hbs.md:92]: **issue (probably user error)** I'm seeing this error 
```
upbound-provider-aws-rds-031cc3ef9f6c-59c7d5df64-vz2lc provider-aws-rds E0616 19:12:03.237587       1 reflector.go:140] k8s.io/client-go@v0.26.3/tools/cache/reflector.go:169: Failed to watch *v1beta1.ProviderConfig: failed to list *v1beta1.ProviderConfig: providerconfigs.aws.upbound.io is forbidden: User "system:serviceaccount:crossplane-system:upbound-provider-aws-rds-031cc3ef9f6c" cannot list resource "providerconfigs" in API group "aws.upbound.io" at the cluster scope
``` 

Did you run into anything similar? I am running on TAP 1.5 if that matters

Comment (Samze) [inline services-toolkit/how-to-guides/dynamic-provisioning-rds.hbs.md:92]: **thought** I followed these instructions and it actually didn't work for me. I had to also include `aws_session_token` in the secret, but maybe thats just how our AWS account is setup through cloudgate. Did you have to do the same?

File: services-toolkit/how-to-guides/dynamic-provisioning-rds.hbs.md [hunk 3]
```diff
@@ -7,15 +7,15 @@ to meet their needs.
...
+[Set up dynamic provisioning of service instances](../tutorials/setup-dynamic-provisioning.hbs.md)
+might help you to better understand the steps presented in this topic.
 
 ## <a id="prereqs"></a> Prerequisites
 
 Before you configure dynamic provisioning, you must have:
 
-- Access to a Tanzu Application Platform cluster v1.5.0 or later.
-- The Tanzu services CLI plug-in v0.6.0 or later.
+- Access to a Tanzu Application Platform cluster v1.6.0 or later.
```
[the remark this answers is not in the document]
Comment (teddyking) [reply services-toolkit/how-to-guides/dynamic-provisioning-rds.hbs.md:17]: I bumped it as I thought I had read somewhere that provider families require v1.12 of crossplane (we bumped from v1.11 to v1.12 in TAP 1.6). However I can't seem to find where I read that now ... Prob best to err on the side of caution and bump anyway?

[the remark this answers is not in the document]
Comment (teddyking) [reply services-toolkit/how-to-guides/dynamic-provisioning-rds.hbs.md:17]: Ah found it: https://docs.upbound.io/upbound-marketplace/providers/#provider-family-requirements, so it is a strict requirement.

File: services-toolkit/how-to-guides/dynamic-provisioning-rds.hbs.md (hunk 1 above)
Comment (teddyking) [reply services-toolkit/how-to-guides/dynamic-provisioning-rds.hbs.md:78]: Yeah I thought that as well, but it's still what is being recommended: https://docs.upbound.io/upbound-marketplace/providers/#using-controllerconfigs

File: services-toolkit/how-to-guides/dynamic-provisioning-rds.hbs.md (hunk 2 above)
Comment (teddyking) [reply services-toolkit/how-to-guides/dynamic-provisioning-rds.hbs.md:92]: Yeah ... I did have to do that as well. I'm not sure if that's a "normal" setup or not, but perhaps I can add a note. We do mention the session key in our older docs here: https://docs.vmware.com/en/Services-Toolkit-for-VMware-Tanzu-Application-Platform/0.9/svc-tlk/usecases-consuming_aws_rds_with_crossplane.html#configure-aws-provider-4

File: services-toolkit/how-to-guides/dynamic-provisioning-rds.hbs.md [hunk 4]
```diff
@@ -7,15 +7,15 @@ to meet their needs.
 
 If you are not already familiar with dynamic provisioning in Tanzu Application Platform,
 following the tutorial
-[Set up dynamic provisioning of service instances](../tutorials/setup-dynamic-provisioning.hbs.md).
-might be help you understand the steps presented in this topic.
+[Set up dynamic provisioning of service instances](../tutorials/setup-dynamic-provisioning.hbs.md)
```
[the remark this answers is not in the document]
Comment (teddyking) [reply services-toolkit/how-to-guides/dynamic-provisioning-rds.hbs.md:9]: Nice, updated the text.

File: services-toolkit/how-to-guides/dynamic-provisioning-rds.hbs.md (hunk 2 above)
Comment (teddyking) [reply services-toolkit/how-to-guides/dynamic-provisioning-rds.hbs.md:91]: Added a note.

**Pull request merged**